core/runtime/src/main/java/io/quarkiverse/langchain4j/RegisterAiService.java

```diff
@@ -97,6 +98,18 @@
     @Deprecated(forRemoval = true)
     Class<? extends Retriever<TextSegment>> retriever() default NoRetriever.class;
 
+    /**
+     * Configures the way to obtain the {@link ContentRetriever}.
+     * The Supplier may or may not be a CDI bean (but most
+     * typically it will, so consider adding a bean-defining annotation to
+     * it). If it is not a CDI bean, Quarkus will create an instance
+     * by calling its no-arg constructor.
+     *
+     * If unspecified, Quarkus will attempt to locate a CDI bean that
```

Comment:
One problem here is that we have exactly the same approach for the `retrievalAugmentor` (that it can be located automatically), but an AI service can only have one of these (not both). And also the same for the classic old `retriever`. An AI service can have at most one out of these **three** components. So we need to decide how to detect and handle potential conflicts. 

The `AiServicesProcessor` already [throws an error](https://github.com/quarkiverse/quarkus-langchain4j/blob/0.9.0/core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java#L256)  if both `retriever` and `retrievalAugmentor` are set. 

But the fact that we have auto-detection makes it a bit harder to detect when an error should be thrown.

Maybe the `AiServicesRecorder` should keep track of what `Retriever`/`ContentRetriever`/`RetrievalAugmentor` instances it creates and if more than one of these exists, throw an error? I think that would be best.

Reply:
Another possible would be to move the conflict detection to Langchain4j and let the `dev.langchain4j.service.AiServices` builder throw an error any two out of `retriever`/`contentRetriever`/`retrievalAugmentor`  get called (instead of silently overriding the previous call, it would throw an error - @langchain4j WDYT?)

Reply:
@jmartisk good point, it makes sense to check this in `AiServices`. BTW, old `retriver` is deprecated now and will be removed soon

Reply:
I've submitted https://github.com/langchain4j/langchain4j/pull/710 so if it's merged, we won't have to be that careful about checking it on the Quarkus side